src/pages/gallery/gallery.tsx

```diff
@@ -6,57 +6,49 @@ import Input from "../../components/design-system/input";
+	sort: SortOrder
+};
 export default function Gallery({ games, tags }: { games: GameMetadata[], tags: string[] }) {
 	const [gamesState, setGamesState] = useState<GalleryGameMetadata[]>([]);
-	const [sortOrder, setSortOrder] = useState<string>("");
-	const [tagFilter, setTagFilter] = useState<string>("");
-	const [searchQuery, setSearchQuery] = useState<string>("");
-
-	function sortGames(games: GameMetadata[], order: SortOrder): void {
-		if (order === "chronological") {
-			games.sort((a, b) => Date.parse(b.addedOn) - Date.parse(a.addedOn))
-				.slice(0, 10)
-				.forEach(game => (game.isNew = true));
-		}
-		if (order === "ascending") games.sort((a, b) => a.title.toLowerCase() > b.title.toLowerCase() ? 1 : -1);
-		if (order === "descending") games.sort((a, b) => b.title.toLowerCase() > a.title.toLowerCase() ? 1 : -1);
-
-		// put tutorials first
-		games.sort((a, _) => a.tags.includes("tutorial") ? -1 : 1)
-	}
-
-	function filterTags(games: GameMetadata[]): void {
-		if (tagFilter === "") {
-			let otherGames = [...games];
-			sortGames(otherGames, sortOrder as SortOrder);
-			setGamesState(otherGames as GalleryGameMetadata[]);
-			return;
-		}
-		setGamesState(games.filter(game => game.tags.includes(tagFilter)) as GalleryGameMetadata[]);
-	}
-
-	useEffect(() => {
-		let otherGames = [...gamesState];
-		sortGames(otherGames, sortOrder as SortOrder)
-		setGamesState(otherGames);
-	}, [sortOrder]);
+	const [filter, setFilter] = useState<Filter>({query: "", sort: SortOrder.CHRONOLOGICAL, tags: [] })
 
-	useEffect(() => { filterTags(games); }, [tagFilter]);
 	useEffect(() => {
-		setGamesState(games.map(game => {
-			let _game: any = { ...game };
-			let query = searchQuery.toLowerCase();
-			const matchesQuery = game.title.toLowerCase().includes(query) ||
+		const lowerCaseQuery = filter.query.toLowerCase();
+		const _games = games.filter(
+				game => game.lowerCaseTitle.includes(lowerCaseQuery) || 
+				game.lowerCaseAuthor.includes(lowerCaseQuery)
+			) // filter by query
+			.filter(game => { // filter by tags
+				for (const tag of filter.tags) {
+					if (game.tags.indexOf(tag) === -1) return false;
+				}
+				return true;
+			});
+		setGamesState(sortGames(_games, filter.sort).map(game => ({ ...game, show: true })) as GalleryGameMetadata[]);
 
-			game.author.toLowerCase().includes(query);
-			_game.show = matchesQuery;
+	}, [filter]);
 
-			return _game;
-		}));
-	}, [searchQuery]);
+	function sortGames(games: GameMetadata[], order: SortOrder): GameMetadata[] {
+		let _games = [...games];
+		if (order === SortOrder.CHRONOLOGICAL) {
+			_games =  _games.sort((a, b) => Date.parse(b.addedOn) - Date.parse(a.addedOn));
+		}
+		if (order === SortOrder.ASCENDING) _games.sort((a, b) => a.lowerCaseTitle > b.lowerCaseTitle ? 1 : -1);
+		if (order === SortOrder.DESCENDING) _games.sort((a, b) => b.lowerCaseTitle > a.lowerCaseTitle ? 1 : -1);
 
+		// put tutorials first
+		_games.sort((a, _) => a.tags.includes("tutorial") ? -1 : 1)
```

Comment:
should tutorials actually always be first?  this has seemed weird to me when using the UI tbh